Simplified/Reader2/Bookmarks/NYPLReadiumBookmark+Compare.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  NYPLReadiumBookmark+R2.swift
+//
+//  Created by Ettore Pasquini on 4/23/20.
+//  Copyright © 2020 NYPL Labs. All rights reserved.
+//
+
+import Foundation
+import R2Shared
+
+extension NYPLReadiumBookmark {
+
+  /// Determines if a given locator matches the location addressed by this
+  /// bookmark.
+  ///
+  /// - Complexity: O(*1*).
+  ///
+  /// - Parameters:
+  ///   - locator: The object representing the given location in `publication`.
+  ///   - publication: The publication the `locator` is referring to.
+  ///
+  /// - Returns: `true` if the `locator`'s position matches the bookmark's.
+  func locationMatches(_ locator: Locator,
+                       inPublication publication: Publication) -> Bool {
+
+    let locatorChapterProgress: Float?
+    if let chapterProgress = locator.locations.progression {
+      locatorChapterProgress = Float(chapterProgress)
+    } else {
+      locatorChapterProgress = nil
+    }
+
+    guard self.href == locator.href else {
+      return false
+    }
+
+    if self.progressWithinChapter =~= locatorChapterProgress {
```

Comment:
We can just `return self.progressWithinChapter =~= locatorChapterProgress` here

Reply:
lol good catch, yes we can and should :) will fix